### PR TITLE
fix: validation error to CLI Error conversion

### LINKED ISSUE #1332: Invalid Error Format
`start` command throws an error in an unexpected format
**Input**
```graphql
schema @upstream(httpCache: "true") {
  query: Query
}
```

**Actual**

```
 INFO File read: ./examples/rest-api.graphql ... ok
ERROR Validation Error
• Parsing failed because of invalid type: string "true", expected a boolean [@upstream, httpCache]
```

**Expected**

```
 INFO File read: ./examples/rest-api.graphql ... ok
ERROR Invalid Configuration
Caused by:
  • Parsing failed because of invalid type: string "true", expected a boolean [at schema.@upstream.httpCache]
```

### PR DESCRIPTION
**Pull request by mehulmathur16**
**Summary:**  
Added the code for converting validation error to CLI error using appropriate from methods implemented in cli/error.rs file.

**Issue Reference(s):**  
/claim #1332
/fixes #1332

image

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **Refactor**
	- Improved error handling by integrating `ValidationError` into the app's error management system.

### CODE REVIEW

**Comment by mehulmathur16** [comment]
@tusharmath @amitksingh1490 Can you please review this PR ?

**Review by tusharmath — APPROVED** [review]

File: src/main.rs [hunk 1]
```diff
@@ -49,12 +50,19 @@ fn main() -> anyhow::Result<()> {
...
+                            let sources = error
+                                .source()
+                                .map(|error| vec![CLIError::new(error.to_string().as_str())])
+                                .unwrap_or_default();
 
-                    CLIError::new(&error.to_string()).caused_by(sources)
+                            CLIError::new(&error.to_string()).caused_by(sources)
+                        }
+                    };
+                    cli_error
```
Comment (tusharmath) [inline src/main.rs:65]: @coderabbitai We could add simple unit tests for error downcasting. Write a function that takes the error that's produced from run_blocking, and checks various downcasting branches. Add mod tests to write some unit tests for those conversions.

Comment (tusharmath) [reply src/main.rs:65]: @coderabbitai suggest actual tests not just comments.

Comment (mehulmathur16) [reply src/main.rs:65]: @coderabbitai Can you write simulate_errors function & some sample cases ?

**Pull request merged**